sqrl-cli/src/main/java/com/datasqrl/cli/AbstractCompileCmd.java

```diff
@@ -110,6 +101,16 @@ protected void execute(ErrorCollector errors, Path snapshotPath, Optional<Path>
     postprocess(packager, getTargetDir(), plan.getLeft(), plan.getRight());
   }
 
+  protected void execute(ErrorCollector errors) throws Exception {
+    // Do nothing by default
+  }
+
+  @Override
+  protected void runInternal(ErrorCollector errors) throws Exception {
+    compile(errors);
+    execute(errors);
+  }
```

Comment:
I ended up doing this to avoid calling `super.execute(...)` in `RunCmd`, `TestCmd`, and `ExecuteCmd` and keep them compatible with their unit tests.